### DISCUSSION TITLE
access parent field in for expression

### DISCUSSION THREAD

**Discussion by agent-mo** [schibsted/jslt#163] [Q&A] [answered]
**Hi, I want to use jslt to transform json , but happen an unsolvable problem.** 

* The input json data like this : 
`{
	"user_id" : "001",
	"friends": [
		{
			"friend_id": "002"
		},
		{
			"friend_id": "003"
		},
		{
			"friend_id": "004"
		}
	]
}`

* Then , what  output json data I expected like the follow : 
`[
	{
		"user_id": "001",
		"friend_id" : "002" 
	},
	{
		"user_id": "001",
		"friend_id" : "003" 
	},
	{
		"user_id": "001",
		"friend_id" : "004" 
	}
]`

* In jslt expression , I use expression of **for** to traverse the array field **friends** : 
` [
	for (.friends) { "user_id": .user_id , "friend_id" : .friend_id }	
]`

* However , the treansform result can't get field **user_id** 
`[{"friend_id":"002"},{"friend_id":"003"},{"friend_id":"004"}]` 

How can I access field **user_id** out of the scope related array field **friends** ?  
Looking forward for your help， thanks

**Comment by larsga** [comment, marked as answer]
The thing here is that if we allow accessing the parent it becomes very difficult to reuse nodes from the input in the output, which means we have to do costly copying. There is a simple workaround: you can capture the user id in a variable before you start the loop: 

```
let user_id = (.user_id)
[ for (.friends) { "user_id": $user_id , "friend_id" : .friend_id } ]
```

**Reply by agent-mo** [reply]
Nice , thank you ! Your answer fit my requirement .
